pkg/clients/azure/azure.go

```diff
@@ -18,26 +18,38 @@ const (
 	CredentialsKeyTenantID       = "tenantId"
 	CredentialsKeyClientCert     = "clientCertificate"
 	CredentialsKeyClientCertPass = "clientCertificatePassword"
+
+	kubeloginCLIFlagServerID = "server-id"
 )
 
+func kubeloginTokenOptionsFromRESTConfig(rc *rest.Config) (*token.Options, error) {
+	opts := &token.Options{}
+
+	// opts are filled according to the provided args in the execProvider section of the kubeconfig
+	// we are parsing serverID from here
+	// add other flags if new login methods are introduced
+	fs := pflag.NewFlagSet("kubelogin", pflag.ContinueOnError)
+	fs.StringVar(&opts.ServerID, kubeloginCLIFlagServerID, "", "Microsoft Entra (AAD) server application id")
+	err := fs.Parse(rc.ExecProvider.Args)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not parse execProvider arguments in kubeconfig")
+	}
+
+	return opts, nil
+}
+
 func WrapRESTConfig(_ context.Context, rc *rest.Config, credentials []byte, _ ...string) error {
 	m := map[string]string{}
 	if err := json.Unmarshal(credentials, &m); err != nil {
 		return err
 	}
 
-	fs := pflag.NewFlagSet("kubelogin", pflag.ContinueOnError)
```

Comment:
@erhancagirici I am less sure how to crossplane consume this rest config: are we expecting the input config is returned from an AAD enabled AKS cluster? Because using kubelogin's SP login method, it's required to have `server-id` set.

Reply:
thanks for your review, I agree with you on the flags parsing pain here. I created an issue to track it: https://github.com/Azure/kubelogin/issues/394 and will see how to expose it in library mode. We can follow up after the change in kubelogin side.